src/main/groovy/edu/oregonstate/mist/api/InfoResource.groovy

```diff
@@ -12,21 +15,28 @@ import javax.ws.rs.core.Response
  */
 @Path('/')
 class InfoResource extends Resource {
+    Logger logger = LoggerFactory.getLogger(InfoResource.class)
+
     private Info info = new Info()
 
     public InfoResource() {
-        def stream = this.class.getResourceAsStream('/build.properties')
-        if (stream == null) {
-            throw new Exception("couldn't load build.properties")
-        }
+        try {
+            def stream = this.class.getResourceAsStream('/build.properties')
+            if (stream == null) {
+                throw new Exception("couldn't load build.properties")
+            }
 
-        def properties = new Properties()
-        properties.load(stream)
+            def properties = new Properties()
+            properties.load(stream)
+
+            info.name = properties.get('name')
+            info.time = Long.parseLong(properties.getProperty('time'))
+            info.commit = properties.get('commit')
+            info.documentation = properties.get('documentation')
+        } catch (Exception e) {
+            logger.error("Resource initialization exception when loading InfoResource build.properties", e)
```

Comment:
How about if we return the server error that is generated by the Resource class? This would allow us to provide a better error message to the user / developer.